File: backend/src/main/java/com/backend/api/drawing/dto/DrawingDetailDto.java
package com.backend.api.drawing.dto;

import com.backend.domain.comment.dto.CommentDto;
import com.backend.domain.drawing.entity.Drawing;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

@Getter
public class DrawingDetailDto extends DrawingListDto {

    private final String title;
    private final String content;
    private final String artist;
    private final boolean nftYn;
    private final boolean memberLiked;
    private final int likesCount;
    private final int viewCount;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Seoul")
    private final LocalDateTime createdAt;

    private final List<CommentDto> comments;

    public DrawingDetailDto(Drawing drawing, boolean memberLiked) {
        super(drawing);
        this.title = drawing.getTitle();
        this.artist = drawing.getArtist();
        this.content = drawing.getContent();
        this.nftYn = drawing.isNftYn();
        this.memberLiked = memberLiked;
        this.likesCount = drawing.getLikesCount();
        this.viewCount = drawing.getViewCount();
        this.createdAt = drawing.getCreatedAt();
        this.comments = drawing.getComments().stream().map(CommentDto::new).collect(Collectors.toList());

    }
}

File: frontend/likloud/app/src/main/java/com/ssafy/likloud/data/model/DrawingDetailDto.kt
package com.ssafy.likloud.data.model

import androidx.lifecycle.MutableLiveData
import com.google.gson.annotations.SerializedName

data class DrawingDetailDto(

    @SerializedName("drawingId")
    val drawingId: Int,

    @SerializedName("memberId")
    val memberId: Int,

    @SerializedName("title")
    val title: String,

    @SerializedName("artist")
    val artist: String,


    @SerializedName("imageUrl")
    val imageUrl: String,

    @SerializedName("viewCount")
    val viewCount: Int,

    @SerializedName("likesCount")
    val likesCount: Int,

    @SerializedName("memberLiked")
    var memberLiked: Boolean,

    @SerializedName("content")
    val content: String,

    @SerializedName("createdAt")
    val createdAt: String,

    @SerializedName("comments")
    val commentList: MutableList<CommentDto>,

    @SerializedName("nftYn")
    val nftYn: Boolean

)